packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -501,7 +501,10 @@ function createArrayNodeProxy(
 			const maybeIndex = asIndex(key, Number.POSITIVE_INFINITY);
 			if (maybeIndex !== undefined) {
 				// For MVP, we otherwise disallow setting properties (mutation is only available via the array node mutation APIs).
-				return false;
+				// To ensure a clear and actionable error experience, we will throw explicitly here, rather than just returning false.
```

Comment:
If we don't do this, the error users see looks like
```shell
'set' on proxy: trap returned falsish for property '5'
```

This isn't particularly actionable, so throwing our own exception with more explicit instructions seems preferable.

Reply:
Agreed :)